### PR TITLE
fix(picker-column): column renders correctly with selected value

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The picker column is mutating the reference object for the column options. This causes the check that compares the transform value against the option's transform value, to skip applying the button transform values and the column items will overlap (on the second open with a selected value).

Issue URL: #17664

## What is the new behavior?

- Always apply the button transform for the option.
- Picker renders correctly with a selected value.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

|Main|PR|
|---|---|
|<video src="https://user-images.githubusercontent.com/13732623/159815305-7924deaa-8886-465c-9793-e6c9282f0900.mp4"></video>|<video src="https://user-images.githubusercontent.com/13732623/159815222-f02f889a-dd92-40b9-83ce-34f43b5a42d8.mp4"></video>|

### CODE REVIEW

**Review by liamdebeasi — APPROVED** [review]
Looks good. Can we post a dev build on the issue thread and collect feedback prior to merging?

File: core/src/components/picker/test/basic/index.html [hunk 1]
```diff
@@ -124,8 +135,6 @@
       await pickerElement.present();
 
       const { data, role } = await pickerElement.onDidDismiss();
```
Comment (amandaesmith3) [inline core/src/components/picker/test/basic/index.html:137]: It looks like these lines aren't needed anymore, could we remove them?

**Comment by gee1k** [comment]
Please merge it, i need it, thanks

**Comment by liamdebeasi** [comment]
@sean-perkins We can merge this once the branch is synced with `main`. I think we gave developers plenty of times to test, and both Amanda and I have found no issues with the PR.

**Pull request merged**